### PR TITLE
Adding context menu to table component

### PR DESCRIPTION
**Pull request by aasimkhan30**
image
On Click/Enter
image
On Escape
image

### CODE REVIEW

File: src/sql/workbench/browser/modelComponents/table.component.ts [hunk 1]
```diff
@@ -483,7 +505,70 @@ export default class TableComponent extends ComponentBase<azdata.TableComponentP
...
+		}
+
+		this._register(
+			this._contextMenuColumns[col.value].onClick((state) => {
+				const cellValue = state.item[col.value];
+				const actions: IAction[] = [];
+				let addSeparator = false;
+				for (const [index, command] of cellValue.commands.entries()) {
+					const isCommand = typeof command === 'string';
+					if (addSeparator || (!isCommand && index !== 0)) {
```
[the remark this answers is not in the document]
Comment (aasimkhan30) [reply src/sql/workbench/browser/modelComponents/table.component.ts:527]: I got the logic from declarative table

File: src/sql/workbench/browser/modelComponents/table.component.ts [hunk 2]
```diff
@@ -483,7 +505,70 @@ export default class TableComponent extends ComponentBase<azdata.TableComponentP
...
+					getActions: () => actions,
+					getActionsContext: () => cellValue.context,
+					onHide: (cancel) => {
+						if (cancel) {
+							/**
+							 * If the context menu is closed by using the escape key we change
+							 * the focus back to the table. In case an option is selected, we
+							 * will let users handle the focus
+							 */
+							this.focus();
```
[the remark this answers is not in the document]
Comment (aasimkhan30) [reply src/sql/workbench/browser/modelComponents/table.component.ts:557]: That means the caller should handle where the focus goes.

[the remark this answers is not in the document]
Comment (Charles-Gagnon) [reply src/sql/workbench/browser/modelComponents/table.component.ts:556]: Yeah, but what "caller"? Are you implying that the extension side should be handling where the focus goes? That doesn't seem great to require that - I imagine most of the time they just want the focus to go back to the grid anyways. Did you borrow this logic from somewhere or was there a specific feature request asking for this?

[the remark this answers is not in the document]
Comment (aasimkhan30) [reply src/sql/workbench/browser/modelComponents/table.component.ts:556]: I did not borrow this part of the code from somewhere. The rest of the code is borrowed from declarative table's context menu column. I will change it to always focus back to the table then.

**Review by Charles-Gagnon — APPROVED** [review]

File: src/sql/workbench/browser/modelComponents/table.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Charles-Gagnon) [reply src/sql/workbench/browser/modelComponents/table.component.ts]: Sounds good. If someone needs to have the focus be elsewhere then we can look into adding that functionality later on. Keeping it simple for now seems like the best option.

**Pull request merged**